wemake_python_styleguide/visitors/ast/statements.py

```diff
@@ -286,3 +289,40 @@ def _check_indentation(
                     elements[index - 1].lineno,
                     multi_line_mode,
                 )
+
+
+@final
+class PointlessStarredVisitor(BaseNodeVisitor):
+    """Responsible for absence of useless starred expressions."""
+    _pointless_star_nodes: ClassVar[AnyNodes] = (
+        ast.Dict,
+        ast.List,
+        ast.Set,
+        ast.Tuple,
+    )
+
+    def visit_Call(self, node: ast.Call) -> None:
+        """Checks useless call arguments"""
+        self._find_starred_empty_args(node.args)
+        self._find_double_starred_empty_dict(node.keywords)
+        self.generic_visit(node)
+
+    def _find_starred_empty_args(
+        self,
+        args: Sequence[ast.AST]
+    ) -> None:
+        for node in args:
+            if isinstance(node, ast.Starred):
+                if isinstance(node.value, self._pointless_star_nodes):
+                    if hasattr(node.value, 'keys') and not node.value.keys:
```

Comment:
We do not use `hasattr`, because it indicates a bad design. What do I suggest instead?

```python
def _is_pointless_star_collection(self, node: ast.AST) -> bool:
     if isinstance(node, (ast.List, ast.Tuple, ...)):
          return not node.elts
     elif isinstance(node, ...)
           return not node.keys
     return False
```

Reply:
The second thought: why do we care about `keys` and `elts` at all? It does not make any difference: `*()` and `*(1, 2, 3)` are both wrong.

Reply:
I misunderstood the task. We forbid all cases like:
```python
print(*(), **{})
print(*(1, 2, 3), **{'end': '\n'})
```
Did I understand correctly?

Reply:
Yes, because:
- `print(*[])` == `print()`
- `print(*[1, 2, 3])` == `print(1, 2, 3)`